sockeye/inference.py

```diff
@@ -1278,13 +1281,16 @@ def _beam_search(self,
 
             # (3) get beam_size winning hypotheses for each sentence block separately
             # TODO(fhieber): once mx.nd.topk is sped-up no numpy conversion necessary anymore.
-            scores = scores.asnumpy()  # convert to numpy once to minimize cross-device copying
+            if not self.use_mxnet_topk:
```

Comment:
lets branch on context here

Reply:
See my comment below. I had made all these changes and forgot to commit them before pushing. I think you'll like it now.